src/plugins/explore/public/plugin.ts

```diff
@@ -215,153 +219,162 @@ export class ExplorePlugin
+        toastNotifications: core.notifications.toasts,
+        stateParams: [
+          {
+            osdUrlKey: '_g',
+            stateUpdate$: setupDeps.data.query.state$.pipe(
+              filter(
+                (value: Record<string, unknown>) =>
+                  !!((value.changes as any)?.time || (value.changes as any)?.refreshInterval)
+              ),
+              map((value: Record<string, unknown>) => ({
+                ...(value.state as Record<string, unknown>),
+                // Note: We don't use data plugin's filterManager, filters are managed in Redux
+              }))
+            ),
+          },
+        ],
+        getHistory: () => {
+          return this.currentHistory!;
+        },
+      });
+      this.stopUrlTrackingCallbackByApp[flavor ?? 'explore'] = stopUrlTracker;
 
-        // If there's no flavor id, by default redirect to the logs flavor.
-        if (!flavor) {
-          coreStart.application.navigateToApp(`${PLUGIN_ID}/${ExploreFlavor.Logs}`, {
-            path: '#/',
-            replace: true,
-          });
-          return () => {};
-        }
+      return {
+        id: PLUGIN_ID,
+        title: PLUGIN_NAME,
+        updater$: appStateUpdater.asObservable(),
+        order: 1000,
+        workspaceAvailability: WorkspaceAvailability.insideWorkspace,
+        euiIconType: 'inputOutput',
+        defaultPath: '#/',
+        category: DEFAULT_APP_CATEGORIES.opensearchDashboards,
+        mount: async (params: AppMountParameters) => {
+          if (!this.initializeServices) {
+            throw Error('Explore plugin method initializeServices is undefined');
+          }
+
+          // Get start services
+          const { core: coreStart, plugins: pluginsStart } = await this.initializeServices();
+          const isExploreEnabledWorkspace = await this.getIsExploreEnabledWorkspace(coreStart);
+          // We want to limit explore UI to only show up under the explore-enabled
+          // workspaces. If user lands in the explore plugin URL in a different
+          // workspace, we will redirect them to classic discover. We will also redirect if
+          // they have manually selected classic discover
+          if (
+            !isExploreEnabledWorkspace ||
+            !!localStorage.getItem(SHOW_CLASSIC_DISCOVER_LOCAL_STORAGE_KEY)
+          ) {
+            coreStart.application.navigateToApp('discover', { replace: true });
+            return () => {};
+          }
+
+          // If there's no flavor id, by default redirect to the logs flavor.
+          if (!flavor) {
+            coreStart.application.navigateToApp(`${PLUGIN_ID}/${ExploreFlavor.Logs}`, {
+              path: '#/',
+              replace: true,
+            });
+            return () => {};
+          }
+
+          this.currentHistory = params.history;
+
+          // make sure the index pattern list is up to date
+          pluginsStart.data.indexPatterns.clearCache();
```

Comment:
should we be usiung `data.dataViews.clearCache()` as indexPatterns is deprecated?

Reply:
I have kept the code as is for this. Will leave it upto the core team if needs to be updated